include/epiworld/model-meat.hpp

```diff
@@ -2706,6 +2720,61 @@ inline void Model<TSeq>::draw(
 
 }
 
+template<typename TSeq>
+inline void Model<TSeq>::get_hospitalizations(
+    std::vector< int > & date,
+    std::vector< int > & virus_id,
+    std::vector< int > & counts
+) const
+{
+    // Clear the output vectors
+    date.clear();
+    virus_id.clear();
+    counts.clear();
+
+    // Get the transition matrix
+    std::vector< std::string > state_from;
+    std::vector< std::string > state_to;
+    std::vector< int > transition_date;
+    std::vector< int > transition_counts;
+
+    db.get_hist_transition_matrix(
+        state_from, state_to, transition_date, transition_counts, true
+    );
+
+    // Filter for transitions TO states containing "Hospitalized"
+    // We need to aggregate by date and virus
+    size_t n_days = get_ndays() + 1u;
+    size_t n_viruses = get_n_viruses();
+
+    // Initialize output vectors
+    date.assign(n_days * n_viruses, 0);
+    virus_id.assign(n_days * n_viruses, 0);
+    counts.assign(n_days * n_viruses, 0);
+
+    // Process transitions
+    for (size_t i = 0u; i < state_to.size(); ++i)
+    {
+        // Check if the destination state contains "Hospitalized"
+        // Using "ospitalized" to match both "Hospitalized" and "Detected Hospitalized"
+        if (state_to[i].find("ospitalized") != std::string::npos)
```

Comment:
Why not extracting the hospitalized states beforehand? That way, mapping the hospitalized state labels to actual numbers, you can skip doing the regex in each step; it should be faster. You can get the states using the fundction `get_states()` (or similar) from the model class.